crypto/dh/dh_key.c

```diff
@@ -230,3 +230,39 @@ static int dh_finish(DH *dh)
     BN_MONT_CTX_free(dh->method_mont_p);
     return 1;
 }
+
+int DH_buf2key(DH *key, const unsigned char *buf, size_t len)
+{
+    BIGNUM *pubkey;
+
+    pubkey = BN_bin2bn(buf, len, NULL);
+    if (!pubkey)
```

Comment:
Use explicit NULL tests.  There are several places to make this change.  Also, for this cluster of ```if``` statements, you could write it like this:
```
if ((pubkey = BN_bin2bn(...)) == NULL
        || BN_is_zero()
        || !DH_set0_key))
    goto err;